### PR TITLE
Leslie's approach

### PR DESCRIPTION
**Pull request by niladridutt**
The basic code is from : https://jovianlin.io/pytorch-with-gpu-in-google-colab/ (I've added changes to reflect Leslie' approach)
The notebook is Colab friendly.
It's not exactly how Leslie approached the problem. Instead of taking one example of each class I've taken examples at random.  Training time is 1/4th the time (7.8 seconds vs 30.8 seconds on Colab) it would take on a normal network. Using this approach I got a test accuracy of about 96% (training the entire dataset is giving an accuracy of 97%).

### CODE REVIEW

**Comment by Gokkulnath** [comment]
@niladri99  Hey !  Thanks for getting us started by sharing your notebook, Notebook looks great!. Neat and Clean implementation. 
I think the main intention of Leslie's idea is to give the model one sample of each classes initially and increase the number of samples eventually. This makes sure that we make the most of the training data and try to avoid over fitting (My perception) without using too much data. 

 In your Implementation, i have a strong feeling that random samples might not always give the model equal number of samples of each class as a result model might not be exposed  to certain classes initially and can have class imbalance in the later stages. we might have to rework on that part.

**Comment by niladridutt** [comment]
Yeah, I got that. I just wanted us to get started and hence I wrote a basic code which can be modified to completely reflect Leslie's approach. I tried this because this was quick and I wanted to see how good the accuracy can get. Cheers!

**Comment by iacolippo** [comment]
Hi guys,

I've pushed a simple script with a helper function that gets a balanced
list of indices from the dataset and an example of how to use it together
with SubsetRandomSampler. Instantiating new dataloaders is cheap, so it
should be kind of efficient. It's a ugly hack and there is probably a
better way to do it, but it works and you can use it for your experiments!

2018-05-21 21:48 GMT+02:00 Niladri Shekhar Dutt <notifications@github.com>:
